flytekit/clis/sdk_in_container/run.py

```diff
@@ -763,6 +767,57 @@ def get_command(self, ctx, name):
         )
 
 
+class YamlFileReadingCommand(click.RichCommand):
+    def __init__(self, name: str, params: typing.List[click.Option], help: str, callback: typing.Callable = None):
+        params.append(
+            click.Option(
+                ["--flyte-inputs-file"],
+                required=False,
+                type=click.Path(exists=True, dir_okay=False, resolve_path=True),
+                help="Path to a YAML | JSON file containing inputs for the workflow.",
+            )
+        )
+        super().__init__(name=name, params=params, callback=callback, help=help)
+
+    def parse_args(self, ctx: Context, args: t.List[str]) -> t.List[str]:
+        if "--flyte-inputs-file" in args:
+            idx = args.index("--flyte-inputs-file")
+            args.pop(idx)
+            f = args.pop(idx)
+            with open(f, "r") as f:
+                import yaml
+
+                try:
+                    inputs = yaml.safe_load(f)
+                except yaml.YAMLError as e:
+                    yaml_e = e
+                    try:
+                        inputs = json.load(f)
+                    except json.JSONDecodeError as e:
+                        click.secho(
```

Comment:
lets raise an exception here - maybe of type click.BadParameterException ?

Reply:
Do you mean changing the `secho` lines (line 797 - 803) to something like this?
```
raise click.BadParameter(
    message=f"Could not load the inputs file. Please make sure it is a valid JSON or YAML file."
    f"\n json error: {e},"
    f"\n yaml error: {yaml_e}",
    param_hint="--flyte-inputs-file",
)
```

Reply:
yes